analyzers/src/SonarAnalyzer.CSharp/Rules/Hotspots/LooseFilePermissions.cs

```diff
@@ -0,0 +1,152 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using SonarAnalyzer.Common;
+using SonarAnalyzer.Extensions;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules.Hotspots
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    [Rule(DiagnosticId)]
+    public sealed class LooseFilePermissions : HotspotDiagnosticAnalyzer
+    {
+        private const string DiagnosticId = "S2612";
+        private const string MessageFormat = "Make sure this permission is safe.";
+        private const string Everyone = "Everyone";
+
+        private static readonly HashSet<string> WeakFileAccessPermissions = new HashSet<string>
+        {
+            "AllPermissions",
+            "DefaultPermissions",
+            "OtherExecute",
+            "OtherWrite",
+            "OtherRead",
+            "OtherReadWriteExecute"
+        };
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        private static readonly DiagnosticDescriptor Rule = DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager).WithNotConfigurable();
+
+        public LooseFilePermissions() : this(AnalyzerConfiguration.Hotspot) { }
+
+        internal LooseFilePermissions(IAnalyzerConfiguration configuration) : base(configuration)
```

Comment:
Styling: This is my preference so no need to tackle if you do not agree. The opening and closing parenthesis can be moved to line 58 as nothing happens in the constructor.